internal/gatewayapi/testdata_test.go

```diff
@@ -0,0 +1,2308 @@
+package gatewayapi
```

Comment:
I can externalize these to YAML files if that's our preferred approach for test data; the main thing I find nice about keeping them in Go is that they're easy to navigate to (i.e. "Go to definition...") from the place where they're used.

Reply:
I prefer yaml i/p & o/p for config A -> config B type translations because its easier to read and write test cases once the infra is set up, this is just my personal preference,  the current approach in go should be fine for now